editor/test/resources/editor_extensions/transact_attachment_project/test.editor_script

```diff
@@ -631,6 +661,93 @@ function M.get_commands()
+                                        id = "char",
+                                        components = {
+                                            {
+                                                type = "sprite",
+                                                position = {0.5, 0.5, 0.5}
+                                            },
+                                            {
+                                                type = "collisionobject",
+                                                shapes = {
+                                                    {
+                                                        id = "box",
+                                                        type = "shape-type-box",
+                                                        dimensions = {2.5, 2.5, 2.5},
+                                                    }
+                                                }
+                                            },
+                                            {
+                                                type = "reference",
+                                                path = "/test.script",
+                                                __num = 37
+                                            }
+                                        }
+                                    }
+                                }
+                            },
+                            {
+                                type = "reference-go",
+                                id = "empty-ref"
+                            }
+                        }
+                    }),
+                    editor.tx.add(coll, "children", {
+                        type = "reference-collection",
+                        id = "empty-collection"
+                    }),
+                    editor.tx.add(coll, "children", {
+                        type = "reference-collection",
+                        path = "/ref.collection"
+                    }),
+                    editor.tx.add(coll, "children", {
+                        type = "reference-collection",
+                        path = "/readonly/readonly.collection"
+                    }),
+                    editor.tx.add(coll, "children", {
+                        type = "reference-go",
+                        path = "/readonly/readonly.go",
+                        children = {
+                            {
+                                type = "go",
+                                id = "allowed-child-of-readonly-go"
+                            }
+                        }
+                    })
+                })
+
+                print("After transaction (add gos and collections):")
+                print_coll(coll)
+
+                print("Transaction: clear collection")
+                editor.transact({ editor.tx.clear(coll, "children") })
+
+                print("After transaction (clear collection)")
+                print_coll(coll)
```

Comment:
I'd like to also see some tests that add referenced and embedded game objects below an already-existing game object in a collection.

Similarly, we're missing tests for adding them to an existing game object located below an existing collection reference. Not really sure if the editor lets you do that, and where it ends up in that case, but I think we need to have a well-defined behavior for it. 

Finally, we should test that you get an error if you try to add or remove stuff from game objects or collections below a read-only directory. Although maybe that's already covered in a resource-agnostic way? There may be special circumstances due to them being different node types, though.